scanpy/tests/test_rank_genes_groups.py

```diff
@@ -139,6 +139,20 @@ def test_results_layers():
     for name in true_scores_t_test.dtype.names:
         assert not np.allclose(true_scores_t_test[name][:7], adata.uns['rank_genes_groups']['scores'][name][:7])
 
+def test_singlets():
+    pbmc = pbmc68k_reduced()
+    pbmc.obs['louvain'] = pbmc.obs['louvain'].cat.add_categories(['11'])
+    pbmc.obs['louvain'][0] = '11' 
+    
+    with pytest.raises(ValueError, match=rf"Could not calculate statistics.*{'11'}"):
+        rank_genes_groups(pbmc, groupby = 'louvain')
+
+@pytest.mark.xfail
+def test_emptycat():
+    pbmc = pbmc68k_reduced()
+    pbmc.obs['louvain'] = pbmc.obs['louvain'].cat.add_categories(['11'])
+    
+    rank_genes_groups(pbmc, groupby = 'louvain')
```

Comment:
```suggestion
def test_emptycat():
    pbmc = pbmc68k_reduced()
    pbmc.obs['louvain'] = pbmc.obs['louvain'].cat.add_categories(['11'])

    with pytest.raises(ValueError, match=rf"Could not calculate statistics.*{'11'}"):
        rank_genes_groups(pbmc, groupby = 'louvain')
```

Reply:
@pinin4fjords, could you add this? I'd do it myself but git is giving me some trouble with pushing to this PR.